cmdstanpy/cmdstan_args.py

```diff
@@ -619,6 +621,28 @@ def validate(self) -> None:
                     ' cannot write to dir: {}'.format(self.output_dir)
                 ) from exc
 
+        if self.sig_figs is not None:
+            if (
+                not isinstance(self.sig_figs, int)
+                or self.sig_figs < 1
+                or self.sig_figs > 18
+            ):
+                raise ValueError(
+                    'sig_figs must be an integer between 1 and 18,'
+                    ' found {}'.format(self.sig_figs)
+                )
+            # sig_figs not valid before version 2.25; allow when version is n/a
+            version = os.path.basename(cmdstan_path())
+            if version.startswith('cmdstan-') and version[8].isdigit():
```

Comment:
Is there a version file somewhere or something we can call with subprocess and parse?

Reply:
yes, we can either grep "CMDSTAN_VERSION" out of the makefile, or call `make print-CMDSTAN_VERSION`